catroid/src/androidTest/java/org/catrobat/catroid/test/stage/StageListenerTest.java

```diff
@@ -0,0 +1,84 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.stage;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import com.badlogic.gdx.graphics.Pixmap;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.TextureRegion;
+
+import org.catrobat.catroid.common.LookData;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.stage.StageListener;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(AndroidJUnit4.class)
+public class StageListenerTest {
+
+	@Test
+	public void testDisposeTexture() {
+		Project project = mock(Project.class);
+		Scene scene = mock(Scene.class);
+		Sprite sprite = mock(Sprite.class, RETURNS_DEEP_STUBS);
+		LookData lookData = mock(LookData.class, RETURNS_DEEP_STUBS);
+		Pixmap pixmap = mock(Pixmap.class);
+		TextureRegion textureRegion = mock(TextureRegion.class);
+		Texture texture = mock(Texture.class);
+
+		List<Scene> scenes = new ArrayList<>();
+		List<Sprite> sprites = new ArrayList<>();
+		List<LookData> lookDatas = new ArrayList<>();
+		scenes.add(scene);
+		sprites.add(sprite);
+		lookDatas.add(lookData);
+
+		when(project.getSceneList()).thenReturn(scenes);
+		when(scene.getSpriteList()).thenReturn(sprites);
+		when(sprite.getLookDataList().size()).thenReturn(lookDatas.size());
+		when(sprite.getLookDataList().get(anyInt())).thenReturn(lookData);
+		when(lookData.getPixmap()).thenReturn(pixmap);
+		when(lookData.getTextureRegion()).thenReturn(textureRegion);
+		when(textureRegion.getTexture()).thenReturn(texture);
+
+		StageListener stageListener = new StageListener(project);
+		stageListener.textureDisposal();
+
+		verify(lookData.getPixmap(), times(1)).dispose();
+		verify(lookData.getTextureRegion().getTexture(), times(1)).dispose();
+	}
+}
```

Comment:
Please remove the test. I tried to test it cleanly but the StageListener in it's current state isn't unit testable.

JFI: you don't need to mock everything :laughing:
If it's easy and fast to create an object, don't mock it. If you have to verify behavior which can't be checked by public interfaces, mock it.
So in the end you just mock `LookData` and verify that `LookData.resetLookData()` is called. You don't need to retest that the resources are freed as it's already tested in `LookDataTest.testPixmapAndTextureRegionDisposal`